src/firecracker/src/main.rs

```diff
@@ -302,6 +307,13 @@ fn main_exitable() -> ExitCode {
             .single_value("api-sock")
             .map(PathBuf::from)
             .expect("Missing argument: api-sock");
+        let mut payload_limit = None;
+        if !arguments.flag_present("payload_size") {
```

Comment:
This check should be:
```suggestion
        if arguments.flag_present("payload_size") {
```